### ISSUE TITLE
CairoMakie cycle ignores linestyle

### ISSUE THREAD

**Issue by ederag** [JuliaPlots/Makie.jl#970]
The following code, adapted from 
https://discourse.julialang.org/t/ann-makie-v-0-13/61522/13 
to better see what `cycle` does,
works only with ` using GLMakie`.

```julia
with_theme(
    palette = (
        color = [:blue, :green],
        linestyle = [:solid, :dash, :dot],
    ),
    Lines = (
        cycle = [:color, :linestyle],
    )) do

    f = Figure()
    Axis(f[1, 1])
	x = [0, 1]
    for i in 1.0:6.0
        lines!(x, [i, i])
    end
    f
end
```
With `using CairoMakie`, the result is
![image](https://user-images.githubusercontent.com/10962809/119234540-c3a21080-bb2e-11eb-8190-4d8ab0b29c5c.png)

while
```julia
lines([0, 1], [1, 1], color=:green, linestyle=:dash)
```
works as expected, so it is not a linestyle issue.

Reversing the cycle order (`linestyle` then `color`) 
yields exactly the same wrong plot under `CairoMakie`, 
while on `GLMakie` it works perfectly
(traverses first `linestyle`, then colors).

```
julia 1.6.1
[e9467ef8] GLMakie v0.3.1
[13f3f980] CairoMakie v0.5.2
```

**Comment by jkrumbiegel** [comment]
Can't reproduce, your example worked for me:

<img width="539" alt="grafik" src="https://user-images.githubusercontent.com/22495855/119507753-63e67800-bd6f-11eb-860f-c32febc0250f.png">

**Comment by ederag** [comment]
Weird. The original post was from [`Pluto`](https://github.com/fonsp/Pluto.jl) `0.14.5`, and it fails.
Back to the REPL, the same code complemented with `save("try.png", f)` works fine, indeed (and fine with `.svg` as well).
Under `Pluto` again, the same code as in the REPL, with a `save("try_pluto.png", f)`, fails
(both the image displayed and the saved image are wrong).
Anything that can be done here, or should it be reported to `Pluto.jl` ?

**Comment by SimonDanisch** [comment]
I'm running into the same problem in documenter. This is super weird and must be a problem in Makie, triggered by some weird global state...
It must be something in the display system or so, since when I run documenter in vscode, it suddenly works, and the only difference seems to be what's on the display stack.

**Comment by SimonDanisch** [comment]
Ok i found it: https://github.com/JuliaPlots/Makie.jl/pull/984

**Comment by ederag** [comment]
Indeed, it is fixed (tested with `CairoMakie-0.5.2`,  `Pluto-0.14.6`), thanks !
